gherkin/php/src/PickleCompiler.php

```diff
@@ -27,9 +27,19 @@
 
 final class PickleCompiler
 {
+    /** @var array<string, PickleStep\Type|null> */
+    private array $pickleStepTypeFromKeyword = [];
+
+    private Step\KeywordType $lastKeywordType = Step\KeywordType::UNKNOWN;
+
     public function __construct(
         private readonly IdGenerator $idGenerator,
     ) {
+        $this->pickleStepTypeFromKeyword[Step\KeywordType::UNKNOWN->name] = PickleStep\Type::UNKNOWN;
```

Comment:
This can either be a single array assignment:

```php
$this->pickleStepTypeFromKeyword = [
   Step\KeywordType::UNKNOWN->name => PickleStep\Type::UNKNOWN,
// ... etc
];
```

or the same literal array moved to where the property is initialised (currently as `[]`)

Reply:
That is what I tried first, but I did not manage to make psalm happy with this actually